<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1536</width>
    <height>714</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>221</width>
      <height>411</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="title">
     <string>CONNECTION SETTINGS</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QComboBox" name="port_cb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="stopBit_cb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>169</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1,5</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="dataBit_cb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>129</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="editable">
      <bool>false</bool>
     </property>
     <property name="currentText">
      <string>5 </string>
     </property>
     <item>
      <property name="text">
       <string>5 </string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6 </string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7 </string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8 </string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>49</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Serial Port</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>168</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop Bits</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>208</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Parity</string>
     </property>
    </widget>
    <widget class="QPushButton" name="connect_butt">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>249</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(0, 255, 0)</string>
     </property>
     <property name="text">
      <string>CONNECT</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>128</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Data Bits</string>
     </property>
    </widget>
    <widget class="QComboBox" name="parity_cb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>209</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>No Parity</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Even Parity</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Odd Parity</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Mark Parity</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Space Parity</string>
      </property>
     </item>
    </widget>
    <widget class="QTextBrowser" name="status_lb">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>300</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QComboBox" name="baud_cb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>89</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>88</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Baudrate</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="cb_show_log">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>380</y>
       <width>161</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>SHOW DEBUG LOG</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_led_tx">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>340</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="lb_led_rx">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>340</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>20</y>
      <width>201</width>
      <height>261</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="title">
     <string notr="true">PID CONTROLLER</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QLineEdit" name="ki_tb">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>76</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="kp_tb">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>45</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>19</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Kp</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="sp_tb">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>138</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>SP (rpm)</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="kd_tb">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>107</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>16</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Ki</string>
     </property>
    </widget>
    <widget class="QPushButton" name="stop_bt">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>180</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>STOP</string>
     </property>
    </widget>
    <widget class="QPushButton" name="dir_bt">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>220</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>REVERSE</string>
     </property>
    </widget>
    <widget class="QPushButton" name="spid_bt">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>START</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Kd</string>
     </property>
    </widget>
   </widget>
   <widget class="QCustomPlot" name="plot" native="true">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>30</y>
      <width>781</width>
      <height>371</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="gb_debugLog">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>310</y>
      <width>391</width>
      <height>331</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="title">
     <string>DEBUG LOG</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QTextBrowser" name="trans_pl_tb">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>171</width>
       <height>221</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="trans_pl_bt">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>290</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QTextBrowser" name="rec_pl_tb">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>51</y>
       <width>171</width>
       <height>221</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>30</y>
       <width>61</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Receive</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rec_pl_bt">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>293</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Transmit</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QCustomPlot" name="plotError" native="true">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>440</y>
      <width>781</width>
      <height>191</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>20</y>
      <width>191</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="title">
     <string>FREQUENCY (Hz)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QLineEdit" name="pwm_tb">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>71</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pwm_bt">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>SET </string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>110</y>
      <width>191</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="title">
     <string>STATUS</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Val</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>70</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Error</string>
     </property>
    </widget>
    <widget class="QPushButton" name="export_bt">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>110</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>EXPORT</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcd_err">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>64</width>
       <height>23</height>
      </rect>
     </property>
     <property name="digitCount">
      <number>6</number>
     </property>
     <property name="segmentStyle">
      <enum>QLCDNumber::Flat</enum>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcd_val">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>30</y>
       <width>64</width>
       <height>23</height>
      </rect>
     </property>
     <property name="smallDecimalPoint">
      <bool>false</bool>
     </property>
     <property name="digitCount">
      <number>6</number>
     </property>
     <property name="segmentStyle">
      <enum>QLCDNumber::Flat</enum>
     </property>
    </widget>
    <widget class="QLabel" name="lb_motor_status">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1536</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header location="global">qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
